giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/aspects/access/RestSecurityAspect.java

```diff
@@ -123,7 +123,14 @@ public Object checkAppTokenAccess(ProceedingJoinPoint joinPoint,
         }
         
         IAppToken appToken = ((IAppToken)holder.tokenContents);
-        String checkerId = identityProviderRegistry.getCheckerId(appToken.getProviderId());
+        String checkerId = null;
+
+        if (appToken.getAuthorizationType() != null && !appToken.getAuthorizationType().isEmpty()) {
+            checkerId = identityProviderRegistry.getCheckerId(appToken.getProviderId() + "_" + appToken.getAuthorizationType());
+        } else {
+            checkerId = identityProviderRegistry.getCheckerId(appToken.getProviderId());
+        }
```

Comment:
I would move the logic of concatenating provider id and authentication type into the identity provider registry. just give ```getCheckerId``` and ```addProvider``` two parameters and let the registry figure out the logic. Sorry, I didn't think of this earlier, but this will make the code much cleaner.